[Release] Version 1.0.0 (#1)
luizmendez/learning-react-with-dogs: components.md

## Components

### App

#### `<App />`

-   Main component of the app
-   Connects with redux store

> -   Get Redux's state and actions as props

Reviewer: Same thing about the apostrophe.

Author: Done.

-   Retrieves the dog list either from localStorage or an API request
-   Renders all the app content
-   Renders Router components
-   Renders MessageHandler component

### Router

#### `<Router />`

-   High Order Component that renders the appropriate component depending on the current location
-   Delegates what to show depending on the current location path
-   Gets the current location path and divides it into path parts

> -   Have three types of routing components:

Reviewer: ```suggestion
-   Has three types of routing components:
```

Author: Done.

>     -   `<Route />`: compares the current path with route path and render its children if there's a match

Reviewer: ```suggestion
    -   `<Route />`: compares the current path with its path prop and renders its children if they match
```
It doesn't make much sense to use route path if I don't already know the component's API

Author: Thanks. Done.

    -   `<Redirect />`: compares the current path with the redirect prop and changes the location if there is a match
    -   `<Otherwise />`: redirects to the given path if the current location doesn't match any registered path
-   The values of Routes, Redirects and Otherwise are registered in the router internals
-   Reads params in the URI to render sub paths (dog/:dogbreed - dog/shiba)
-   Creates a listener on history.pushState to be able to manage routing without having to reload the page
-   Uses **React Context** to manage and share state

#### `<Route />`

-   Register a path on the router
-   Compares the current path with the route path and if there's a match render the route children 

#### `<Redirect />`

-   Register a redirect path on the router
-   Compares current path with redirect route path and if there's a match history push to the redirection

#### `<Otherwise />`

-   Register a fallback path on the router

#### `<Link />`

-   Renders a link `<a></a>` element enhanced with the processed routes
-   Gets current location from react context

> -   Checks if the href attribute of the link is internal or complete URL

Reviewer: ```suggestion
-   Checks if the href attribute of the link is internal or external
```

Author: Done.

-   If the href is internal calls for a history.pushState

> -   If the href is a complete URL (mostly external links) sets `window.location` or calls `window.open()` depending on the link's target attribute

Reviewer: ```suggestion
-   If the href is external sets `window.location` or calls `window.open()` depending on the link's target attribute
```

Author: Done.
